pkg/util/command_util.go

```diff
@@ -372,48 +373,45 @@ func GetUIDAndGIDFromString(userGroupString string, fallbackToUID bool) (uint32,
 		groupStr = userAndGroup[1]
 	}
 
-	if reflect.TypeOf(userStr).String() == "int" {
-		return 0, 0, nil
-	}
-
-	uidStr, gidStr, err := GetUserFromUsername(userStr, groupStr, fallbackToUID)
-	if err != nil {
-		return 0, 0, err
-	}
-
-	// uid and gid need to be fit into uint32
-	uid64, err := strconv.ParseUint(uidStr, 10, 32)
+	// checkif userStr is a valid id
+	uid, err := strconv.ParseUint(userStr, 10, 32)
 	if err != nil {
-		return 0, 0, err
+		return getIdsFromUsernameAndGroup(userStr, groupStr, fallbackToUID)
 	}
 
-	gid64, err := strconv.ParseUint(gidStr, 10, 32)
+	gid, err := strconv.ParseUint(groupStr, 10, 32)
 	if err != nil {
-		return 0, 0, err
+		// check if the group is specified as a string
+		group, err := getGroupFromStr(groupStr)
+		if err != nil && !fallbackToUID {
```

Comment:
this conditional is too broad. If the error is not nil, but fallbackToUID is true, group will be nil and the program will panic: (see https://github.com/GoogleContainerTools/kaniko/runs/6558115558?check_suite_focus=true#step:6:157)